File: CPP/7. Reverse Integer.cpp
#include <iostream>
#include <cmath>

using namespace std;

class Solution {
public:
    int reverse(int x) {
        bool flag = false;

        if (x < 0) {
            x = abs(x);
            flag = true;
        }

        int minVal = numeric_limits<int>::min();
        int maxVal = numeric_limits<int>::max();

        int rx = 0;

        while (x > 0) {
            int rest = x % 10;

            if (rx > (maxVal - rest) / 10) {
                return 0;
            }

            rx = rx * 10 + rest;            
            x = x / 10;
        }

        if (!flag) {
            return rx;
        }

        return rx * -1;

    }
};

int main() {
    int x = 1534236469;
    Solution sol;
    cout << sol.reverse(x) << endl;
}

File: Python/7. Reverse Integer.py
class Solution:
    def reverse(self, x: int) -> int:
        rx = str(x)[::-1]
        if rx[-1] == '-':
            rx = '-'+rx[:-1]
        rx = int(rx)
        return rx if (rx >= -2**31) and (rx < 2**31) else 0

if __name__ == '__main__':
    x = 1534236469
    result = Solution().reverse(x)
    print(result)